### PR TITLE
test presenter

### PR DESCRIPTION
**Pull request by vutrongthien2408**

### CODE REVIEW

File: app/src/main/java/com/framgia/movie/data/source/MovieTrailerRepository.java [hunk 1]
```diff
@@ -9,8 +9,7 @@
  * Created by TrongThien on 9/29/2017.
  */
 
-public final class MovieTrailerRepository
-        implements TheMovieRemoteDataSource.MovieTrailerDataSource {
+public class MovieTrailerRepository implements TheMovieRemoteDataSource.MovieTrailerDataSource {
```
Comment (DoanVanToan) [inline app/src/main/java/com/framgia/movie/data/source/MovieTrailerRepository.java]: WARNING: Class MovieTrailerRepository should be declared as final.

File: app/src/test/java/com/framgia/movie/MoviePresenterTest.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+                    @Override
+                    public Scheduler apply(Callable<Scheduler> schedulerCallable) throws Exception {
+                        return Schedulers.trampoline();
+                    }
+                });
+        mPresenter.setViewModel(mViewModel);
+    }
+
+    @Test
+    public void getListMovie_shouldReturnTrue_whenGetListMovie() {
```
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/MoviePresenterTest.java]: ERROR: [MethodNamingConventions] Method names should not contain underscores
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#MethodNamingConventions

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+                    @Override
+                    public Scheduler apply(Callable<Scheduler> schedulerCallable) throws Exception {
+                        return Schedulers.trampoline();
+                    }
+                });
+        mPresenter.setViewModel(mViewModel);
+    }
+
+    @Test
+    public void getTrailer_shouldReturnTrue_whenGetTrailer() throws Exception {
```
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: ERROR: [MethodNamingConventions] Method names should not contain underscores
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#MethodNamingConventions

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java [hunk 4]
```diff
@@ -0,0 +1,65 @@
+package com.framgia.movie;
+
+import android.content.Context;
```
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: ERROR: Unused import: android.content.Context. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+import com.framgia.movie.data.model.MovieTrailer;
```
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: ERROR: Unused import: com.framgia.movie.data.model.MovieTrailer. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+import com.framgia.movie.data.source.MovieTrailerRepository;
```
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: ERROR: Unused import: com.framgia.movie.data.source.MovieTrailerRepository. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+import com.framgia.movie.screen.video_trailer.VideoTrailerPresenter;
```
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: ERROR: Unused import: com.framgia.movie.screen.video_trailer.VideoTrailerPresenter. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+import com.framgia.movie.screen.video_trailer.VideoTrailerViewModel;
```
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: ERROR: Unused import: com.framgia.movie.screen.video_trailer.VideoTrailerViewModel. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+import io.reactivex.Observable;
```
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: ERROR: Unused import: io.reactivex.Observable. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+import io.reactivex.Scheduler;
```
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: ERROR: Unused import: io.reactivex.Scheduler. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java [hunk 11]
```diff
@@ -0,0 +1,65 @@
...
+import io.reactivex.android.plugins.RxAndroidPlugins;
```
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: ERROR: Unused import: io.reactivex.android.plugins.RxAndroidPlugins. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java [hunk 12]
```diff
@@ -0,0 +1,65 @@
...
+import io.reactivex.functions.Function;
```
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: ERROR: Unused import: io.reactivex.functions.Function. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+import io.reactivex.schedulers.Schedulers;
```
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: ERROR: Unused import: io.reactivex.schedulers.Schedulers. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java [hunk 14]
```diff
@@ -0,0 +1,65 @@
...
+import java.util.ArrayList;
```
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: ERROR: Unused import: java.util.ArrayList. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java [hunk 15]
```diff
@@ -0,0 +1,65 @@
...
+import java.util.List;
```
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: ERROR: Unused import: java.util.List. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java [hunk 16]
```diff
@@ -0,0 +1,65 @@
...
+import java.util.concurrent.Callable;
```
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: ERROR: Unused import: java.util.concurrent.Callable. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java [hunk 17]
```diff
@@ -0,0 +1,65 @@
...
+import org.junit.Before;
```
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: ERROR: Unused import: org.junit.Before. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java [hunk 18]
```diff
@@ -0,0 +1,65 @@
...
+import org.junit.Test;
```
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: ERROR: Unused import: org.junit.Test. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java [hunk 19]
```diff
@@ -0,0 +1,65 @@
...
+import org.junit.runner.RunWith;
```
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: ERROR: Unused import: org.junit.runner.RunWith. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java [hunk 20]
```diff
@@ -0,0 +1,65 @@
...
+import org.mockito.ArgumentMatchers;
```
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: ERROR: Unused import: org.mockito.ArgumentMatchers. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java [hunk 21]
```diff
@@ -0,0 +1,65 @@
...
+import org.mockito.InjectMocks;
```
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: ERROR: Unused import: org.mockito.InjectMocks. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java [hunk 22]
```diff
@@ -0,0 +1,65 @@
...
+import org.mockito.Mock;
```
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: ERROR: Unused import: org.mockito.Mock. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java [hunk 23]
```diff
@@ -0,0 +1,65 @@
...
+import org.mockito.Spy;
```
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: ERROR: Unused import: org.mockito.Spy. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java [hunk 24]
```diff
@@ -0,0 +1,65 @@
...
+import org.mockito.junit.MockitoJUnitRunner;
```
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: ERROR: Unused import: org.mockito.junit.MockitoJUnitRunner. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java [hunk 25]
```diff
@@ -0,0 +1,65 @@
...
+import static org.mockito.Mockito.verify;
```
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: ERROR: Unused import: org.mockito.Mockito.verify. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java [hunk 26]
```diff
@@ -0,0 +1,65 @@
...
+import static org.mockito.Mockito.when;
```
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: ERROR: Unused import: org.mockito.Mockito.when. Use Ctrl+Shift+O (Eclipse) or Ctrl+Alt+O (Intellij) to sort imports

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java [hunk 27]
```diff
@@ -0,0 +1,65 @@
...
+//    @Mock
+//    private MovieTrailerRepository mRepository;
+//
+//    @Before
+//    public void setup() {
+//        RxAndroidPlugins.setInitMainThreadSchedulerHandler(
+//                new Function<Callable<Scheduler>, Scheduler>() {
+//
+//                    @Override
+//                    public Scheduler apply(Callable<Scheduler> schedulerCallable) throws Exception {
```
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: ERROR: Line is longer than 100 characters (found 102).

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java (hunk 4 above)
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: WARNING: [UnusedImports] Avoid unused imports such as 'android.content.Context'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java (hunk 5 above)
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: WARNING: [UnusedImports] Avoid unused imports such as 'com.framgia.movie.data.model.MovieTrailer'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java (hunk 6 above)
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: WARNING: [UnusedImports] Avoid unused imports such as 'com.framgia.movie.data.source.MovieTrailerRepository'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java (hunk 7 above)
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: WARNING: [UnusedImports] Avoid unused imports such as 'com.framgia.movie.screen.video_trailer.VideoTrailerPresenter'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java (hunk 8 above)
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: WARNING: [UnusedImports] Avoid unused imports such as 'com.framgia.movie.screen.video_trailer.VideoTrailerViewModel'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java (hunk 9 above)
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: WARNING: [UnusedImports] Avoid unused imports such as 'io.reactivex.Observable'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java (hunk 10 above)
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: WARNING: [UnusedImports] Avoid unused imports such as 'io.reactivex.Scheduler'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java (hunk 11 above)
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: WARNING: [UnusedImports] Avoid unused imports such as 'io.reactivex.android.plugins.RxAndroidPlugins'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java (hunk 12 above)
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: WARNING: [UnusedImports] Avoid unused imports such as 'io.reactivex.functions.Function'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java (hunk 13 above)
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: WARNING: [UnusedImports] Avoid unused imports such as 'io.reactivex.schedulers.Schedulers'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java (hunk 14 above)
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: WARNING: [UnusedImports] Avoid unused imports such as 'java.util.ArrayList'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java (hunk 15 above)
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: WARNING: [UnusedImports] Avoid unused imports such as 'java.util.List'
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/imports.html#UnusedImports

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java (hunk 4 above)
Comment (vutrongthien2408) [reply app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: used import

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java (hunk 5 above)
Comment (vutrongthien2408) [reply app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: used import

File: app/src/test/java/com/framgia/movie/TrailerPresenterTest.java (hunk 6 above)
Comment (vutrongthien2408) [reply app/src/test/java/com/framgia/movie/TrailerPresenterTest.java]: used import

File: app/src/test/java/com/framgia/movie/MovieTrailerPresenterTest.java [hunk 28]
```diff
@@ -0,0 +1,65 @@
...
+                    @Override
+                    public Scheduler apply(Callable<Scheduler> schedulerCallable) throws Exception {
+                        return Schedulers.trampoline();
+                    }
+                });
+        mPresenter.setViewModel(mViewModel);
+    }
+
+    @Test
+    public void getTrailer_shouldReturnTrue_whenGetTrailer() throws Exception {
```
Comment (DoanVanToan) [inline app/src/test/java/com/framgia/movie/MovieTrailerPresenterTest.java]: ERROR: [MethodNamingConventions] Method names should not contain underscores
https://pmd.github.io/pmd-5.6.1/pmd-java/rules/java/naming.html#MethodNamingConventions

**Pull request merged**